src/interfaces.jl

```diff
@@ -110,3 +122,27 @@ function reverse_time(sol::Solver)
     options.stopevent = (t,y)->stopevent(2*t0-t,y)
     return solve(ode_reversed,stepper,options)
 end
+
+
+
+# The the elements of tspan should basically be scalars and support
+# most of the scalar operations.  In particular the element type
+# should be closed under the division.
+function make_consistent{S<:AbstractStepper}(y0, tspan::AbstractVector, stepper::Type{S})
+    t_test = 1/(tspan[end]-tspan[1])
+    Tt = typeof(t_test)
+    t_new = convert(Vector{Tt},tspan)
+
+    y_test = y0./t_test
+    Ty = typeof(y_test)
+    if typejoin(Ty,AbstractArray) == AbstractArray
```

Comment:
I'm pretty sure that `typejoin(Ty,AbstractArray) == AbstractArray` is equivalent to `Ty<:AbstractArray`.  The typejoin will produce an `AbstractArray` if `Ty` is a subtype of `AbstractArray` or `Any` otherwise.  I just think that `Ty<:AbstractArray` is more common/readable/standard.  The reason to use `AbstractVector` is that I think we should treat everything but a `AbstractVector` as scalar-like.